### PR TITLE
API route for /spawn

### PR DESCRIPTION
**Pull request by zischwartz**
This PR adds an an API route for spawn (`POST` to `/api/spawn`) so that Thebe can use that endpoint to spawn containers in a way that works across browsers.

In order for this to work, I had to disable xsrf. 

I also added the CORS setting to the /stats endpoint as well, which really should have been in that last PR.

### CODE REVIEW

**Comment by rgbkrk** [comment]
Thanks!

I'm happy with bringing in the `APISpawnHandler` for programmatic usage, but we have to keep the `SpawnHandler` redirect for currently running sites.

File: orchestrate.py [hunk 1]
```diff
@@ -70,12 +76,16 @@ def get(self, path=None):
                 yield self.pool.adhoc(user)
 
                 url = path
-
             app_log.debug("Redirecting [%s] -> [%s].", self.request.path, url)
-            self.redirect(url, permanent=False)
+            # Instead of redirecting, lets just send back the url
+            # Because CORS with redirects is hellish and doesn't work cross browsers
```
Comment (rgbkrk) [inline orchestrate.py]: Good to know.

File: orchestrate.py [hunk 2]
```diff
@@ -70,12 +76,16 @@ def get(self, path=None):
...
+            self.write({'url': url})
+            # self.redirect(url, permanent=False)
+
         except spawnpool.EmptyPoolError:
             app_log.warning("The container pool is empty!")
-            self.render("full.html", cull_period=self.cull_period)
+            self.write({'status': 'full'})
+            # self.render("full.html", cull_period=self.cull_period)
```
Comment (rgbkrk) [inline orchestrate.py]: The changes here in `SpawnHandler` should be dropped.

**Comment by zischwartz** [comment]
Oh, yeah, that was a mistake, I meant to revert those.

**Comment by zischwartz** [comment]
OK, this removes those changes to `SpawnHandler`. Should be good to go now.

**Comment by rgbkrk** [comment]
:shipit:

Thanks!

**Comment by fermion** [comment]
@zischwartz does this PR obviate the need to use https://github.com/zischwartz/tmpnb alongside thebe?

Thanks for your work on this!

**Comment by rgbkrk** [comment]
Yup, that's how the thebe on https://lambdaops.com/share-cell/ works. It's using jupyter/tmpnb for the backend.

**Comment by fermion** [comment]
Thanks @rgbkrk!

**Pull request merged**